evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -126,10 +127,27 @@ def __init__(
         self.allowed_pipelines = []
 
         self.features = features
-        self.allowed_component_graphs = allowed_component_graphs
         self._set_additional_pipeline_params()
         self.exclude_featurizers = exclude_featurizers
 
+        if allowed_component_graphs not in (None, []):
+            if excluded_model_families not in (None, []):
```

Comment:
I'm curious about your decision here to include [] as a possible argument, it's not something we do in other places afaik. It feels a bit clunky, I would certainly prefer to keep `None` as the only no-op option until absolutely necessary if possible.

Reply:
There's also some inconsistency with this in `automl_search.py` w.r.t line 603 and 618, we should clean that up to make it consistent as well!